File: backend/src/ToDoList.Core/Dtos/SyncResponseDto.cs
using System;
using System.Collections.Generic;

namespace ToDoList.Core.Dtos
{
    public class SyncResponseDto
    {
        public DateTime ServerTime { get; set; }
        public List<TodoListDto> Lists { get; set; } = new();
        public List<CategoryDto> Categories { get; set; } = new();
        public List<TodoItemDto> Items { get; set; } = new();
        public List<long> DeletedItemIds { get; set; } = new();
        public List<long> DeletedListIds { get; set; } = new();
        public List<long> DeletedCategoryIds { get; set; } = new();
    }
}

File: clients/android/app/src/main/java/com/yeplist/app/data/remote/dto/SyncResponseDto.kt
package com.yeplist.app.data.remote.dto

data class SyncResponseDto(
    val serverTime: String,
    val lists: List<TodoListDto> = emptyList(),
    val categories: List<CategoryDto> = emptyList(),
    val items: List<TodoItemDto> = emptyList(),
    val deletedItemIds: List<Long> = emptyList(),
    val deletedListIds: List<Long> = emptyList(),
    val deletedCategoryIds: List<Long> = emptyList()
)
